blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractRunImpl.java

```diff
@@ -186,7 +224,21 @@ public BlueTestResultContainer getTests() {
 
     @Override
     public BlueTestSummary getTestSummary() {
-        return BlueTestResultFactory.resolve(run, this).summary;
+        if (getStateObj() == BlueRunState.FINISHED) {
+            try {
+                return TEST_SUMMARY.get(run.getExternalizableId(), new Callable<Optional<BlueTestSummary>>() {
+                    @Override
+                    public Optional<BlueTestSummary> call() throws Exception {
+                        BlueTestSummary summary = BlueTestResultFactory.resolve(run, parent).summary;
+                        return summary == null ? Optional.<BlueTestSummary>absent() : Optional.of(summary);
+                    }
+                }).orNull();
+            } catch (ExecutionException e) {
+                return null;
```

Comment:
should log error message or else the checked exception thrown by value loaded callable will be silently ignored.